pelican/plugins/read_more/read_more.py

```diff
@@ -0,0 +1,86 @@
+"""
+Read More
+=========
+
+This plugin inserts an inline "Read More" link into summary's last HTML element.
+"""
+
+from pelican import contents, signals
+from pelican.generators import ArticlesGenerator
+from pelican.utils import truncate_html_words
+
+try:
+    from lxml.etree import ParserError
+    from lxml.html import fragment_fromstring, fragments_fromstring, tostring
+except ImportError:
+    raise Exception("Unable to find lxml. Read More requires lxml to be installed.")
+
+
+def insert_into_last_element(html, element):
+    """
+    Function to insert an HTML element into another HTML fragment.
+    example:
+        html = '<p>paragraph1</p><p>paragraph2...</p>'
+        element = '<a href="/read-more/">read more</a>'
+        ---> '<p>paragraph1</p><p>paragraph2...<a href="/read-more/">read more</a></p>'
+    """
+    try:
+        item = fragment_fromstring(element)
+    except (ParserError, TypeError):
+        item = fragment_fromstring("<span></span>")
+
+    try:
+        doc = fragments_fromstring(html)
+        doc[-1].append(item)
+
+        return "".join(tostring(e) for e in doc)
+    except (ParserError, TypeError):
+        return ""
+
+
+def insert_read_more_link(instance):
+    """
+    Insert an inline "Read More" link into the last element of the summary
+    :param instance:
+    :return:
+    """
+
+    # only deals with Article type
+    if type(instance) != contents.Article:
+        return
+
+    SUMMARY_MAX_LENGTH = instance.settings.get("SUMMARY_MAX_LENGTH")
+    READ_MORE_LINK = instance.settings.get("READ_MORE_LINK", None)
```

Comment:
Add default values so the Plugin will be functional even without peliacnconf.py adjustment?

```
SUMMARY_MAX_LENGTH = instance.settings.get("SUMMARY_MAX_LENGTH", 50)
READ_MORE_LINK = instance.settings.get("READ_MORE_LINK", "<span>continue</span>")
```